### PR TITLE
refactor: move BLS and KZG specific types to their specific modules

### LINKED ISSUE #849: Move BLS and KZG specific types to their specific modules
We currently have some types that are used in the `Bls` and `Kzg` modules that should be moved to those modules, for example:

https://github.com/lambdaclass/lambda_ethereum_consensus/blob/9445b459c3ab539b4d6e7337e3c98e24a2586853/lib/kzg.ex#L41-L45

https://github.com/lambdaclass/lambda_ethereum_consensus/blob/9445b459c3ab539b4d6e7337e3c98e24a2586853/lib/bls.ex#L23

### PR DESCRIPTION
**Pull request by guha-rahul**
**Motivation**

**Description**

Closes #849

### CODE REVIEW

File: lib/bls.ex [hunk 1]
```diff
@@ -5,6 +5,10 @@ defmodule Bls do
   use Rustler, otp_app: :lambda_ethereum_consensus, crate: "bls_nif"
 
   @type privkey() :: <<_::256>>
+  @type bytes48 :: <<_::384>>
+  @type bytes96 :: <<_::768>>
+  @type bls_pubkey :: bytes48
+  @type bls_signature :: bytes96
```
Comment (MegaRedHand) [inline lib/bls.ex:11]: We should remove the `bls_` prefix. Also, let's inline the `bytesN` types.

File: lib/kzg.ex [hunk 2]
```diff
@@ -4,6 +4,11 @@ defmodule Kzg do
   """
   use Rustler, otp_app: :lambda_ethereum_consensus, crate: "kzg_nif"
 
+  # Max size: BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB
+  @type blob :: binary
+  @type kzg_commitment :: bytes48
+  @type kzg_proof :: bytes48
```
Comment (MegaRedHand) [inline lib/kzg.ex:10]: We should remove the `kzg_` prefix, and inline the `bytesN` types

File: lib/types/types.ex [hunk 3]
```diff
@@ -44,8 +44,4 @@ defmodule Types do
   @type withdrawal_index :: uint64
   @type payload_id :: bytes8
   @type blob_index :: uint64
-  # Max size: BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB
-  @type blob :: binary
```
Comment (MegaRedHand) [inline lib/types/types.ex:48]: Let's keep these in here, to keep changes to a minimum.

File: lib/kzg.ex [hunk 4]
```diff
@@ -4,6 +4,11 @@ defmodule Kzg do
...
+
   @spec blob_to_kzg_commitment(Types.blob()) :: {:ok, Types.kzg_commitment()} | {:error, binary()}
```
Comment (MegaRedHand) [inline lib/kzg.ex:12]: These typespecs should be updated with the new types:


```suggestion
  @spec blob_to_kzg_commitment(blob()) :: {:ok, commitment()} | {:error, binary()}
```

(same for `Bls` functions)

**Review by MegaRedHand — CHANGES_REQUESTED** [review]
Thanks for your contribution!

File: lib/types/types.ex (hunk 3 above)
Comment (guha-rahul) [reply lib/types/types.ex:48]: So i will be importing only the blob type from types.ex.

File: lib/kzg.ex (hunk 4 above)
Comment (guha-rahul) [reply lib/kzg.ex]: updated all the types except blob()

**Comment by guha-rahul** [comment]
@MegaRedHand hey i forgot to the update the typespaces in other files , but even after replacing them , I am getting the `dialyzer` error.

File: lib/types/types.ex [hunk 5]
```diff
@@ -46,6 +46,4 @@ defmodule Types do
   @type blob_index :: uint64
   # Max size: BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB
   @type blob :: binary
-  @type kzg_commitment :: bytes48
-  @type kzg_proof :: bytes48
```
Comment (MegaRedHand) [inline lib/types/types.ex:50]: The dialyzer errors are due to these two guys dissapearing. Let's keep them, but as a reference to `Kzg` types (i.e. `@type kzg_proof :: Kzg.proof()`). Same for Bls.

**Review by MegaRedHand — APPROVED** [review]
Great work!

**Comment by guha-rahul** [comment]
Thank you for helping out!

**Pull request merged**